app/assets/javascripts/accordion.js

```diff
@@ -6,7 +8,205 @@ document.addEventListener('DOMContentLoaded', function (event) {
+    var validationLabel = document.getElementById("publication_content_genre").selectedOptions[0].label;
+    // console.log("validating " + validationLabel);
+
+    switch (validationLabel) {
+        case 'Article': {
+
+            var pubverCheck = document.getElementById("publication_publisher_version").selectedOptions[0].label;
+            if (pubverCheck == ' ') {
+                var pubverLabel = "Publisher Version";
+                validateModal(pubverLabel);
+                return false;
+            }
+            var parTitle = document.getElementById("publication_parent_title").value;
+            if (!parTitle) {
+                var parTitleLabel = "Parent Title";
+                validateModal(parTitleLabel);
+                return false;
+            }
+
+            if (pubverCheck != ' ' && parTitle) {
+                removeModal();
+                console.log("form filled out correctly");
+                return true;
+            }
+
+            alert("Please contact LTDS");
+            break;
+        }
+
+        case 'Book': {
+            console.log("Validating: Book");
+            removeModal();
+            break;
+        }
+
+        case 'Book Chapter': {
+            console.log('Validating: Book Chapter');
+            removeModal();
+            break;
+        }
+
+        case 'Conference Paper': {
+            console.log('Validating: Conference Paper');
+            removeModal();
+            break;
+        }
+
+        case 'Poster': {
+            console.log('Validating: Conference Paper');
+            removeModal();
+            break;
+        }
+
+        case 'Presentation': {
+            console.log('Validating: Presentation');
+            removeModal();
+            break;
+        }
+
+        case 'Report': {
+            console.log('Selected: Report');
+            removeModal();
+            break;
+        }
+
+        case ' ': {
+            console.log('Selected: Report');
+            removeModal();
+            break;
+        }
+    }
+}
+
+function validateModal(x) {
+    // console.log("var x is typeof: " + typeof (x) + " var x value: " + x);
+
+    var modal = document.getElementById("publication_content_genre");
+    var text = '<div class="modal pubform" id="pubvalidatemodal"><div class="modal-dialog mo,dal-dialog-centered"><div class=modal-content><div class=modal-header><h4 class=modal-title>Missing Entries</h4><button class=close data-dismiss=modal type=button>×</button></div><div class=modal-body>' + x + ' is required.<br>Located under Publication Information.</div><div class=modal-footer><button class="btn btn-danger"data-dismiss=modal type=button>Close</button></div></div></div></div>';
+    modal.insertAdjacentHTML("afterend", text);
+    return false;
+
+}
+
+function removeModal() {
+    console.log("modal remove");
+    var modal = document.getElementsByClassName("modal pubform");
+    console.log(modal);
+
+    if (modal) {
+        while (modal[0]) {
```

Comment:
is there a more precise way to do this, perhaps without the use of a while loop? I would prefer an approach where we select all relevant nodes and iterate over a precise list, using `document.querySelectorAll` for instance.

Reply:
using jquery instead `$(".modal.pubform").remove();`